### PR TITLE
0.8 cleanups (0.8.11 release candidate)

### PR DESCRIPTION
**Pull request by GriceTurrble**
## Major Changes

- **Version bumped to 0.8.11**
- **Small update for Python 3.9 compatibility**
  - Removed deprecated call to `ElementTree.getchildren`, removed in 3.9.
  - Code changed to match `develop` version: node is iterated directly.
- **Consolidated tooling compared to `develop` version**
  - Added GitHub Actions test workflow, covering Python 3.5 (as legacy) to Python 3.9 (current)
    - Test suite in the project is minimal compared to `develop`, but is enough to point out some glaring problems.
  - `.converagerc` and `pytest.ini` removed, settings added to `setup.cfg`
    - Will bring this change to `develop` as well.
  - Other `setup.cfg` configs coped from `develop`.
  - Dropped TravisCI config (moving away from Travis in favor of Github Actions workflows)
  - Removed root Makefile (will rely on native Python tooling, rather than `make` aliases)
  - Added `pre-commit`
- **Utilities backported from 1.0dev as `future_utils`**
  - All objects available in 1.0dev under `mws.utils` can be found here as `mws.future_utils` (avoids name conflict with existing `utils.py`)
  - XML utilities that rely on `xmltodict` dependency (not added to `master` branch version) will raise `NotImplementedError` instead.
- **Parameter cleaning updated to match `develop` version**
  - Corrects for #214 error (part of the reason for the utils backporting, just for completeness)
- **Old duplicate utilities removed**
  - `remove_empty`, replaced by future util `remove_empty_params`
  - `remove_namespace`, replaced by future util `remove_xml_namespaces`
  - `MWS.get_timestamp`, deprecated, replaced by future util `mws_utc_now`
    - Deprecated `get_timestamp` would return a formatted string. This updated version uses `mws_utc_now` (which returns a `datetime.datetime` object) and then explicitly returns `.isoformat()` output, so the output is the same as before
    - Code that used `get_timestamp` now uses `mws_utc_now`, which returns `datetime.datetime` instances. Other backported utilities perform proper parameter cleaning (similar to `develop`) to convert to `str` output as needed.
- **`ObjectDict` backported from 1.0dev**
  - "Old" version of the class did not have parity with the "new" one (though the new one is marked for deprecation). Native iteration was not possible as expected in the newer API.
  - This brings parity between the `master` and `develop` versions of that particular class.

## Minor changes

- `requirements-dev.txt` added, copied from `develop` version
- `requirements.txt` updated to match that from `develop` version
- `setup.py` now correctly uses `setuptools.find_packages()`, similar to `develop` version.
- Package version corrected throughout project (found some spots where it was still 0.7!)
  - User agent string should correctly import the project version to automatically update itself for later releases.
- LICENSE file changed to plain file, no `.txt` extension (MANIFEST updated to match)
- All files run through `black` per `pre-commit` full run
- Some files touched by `pre-commit` full run for line endings and trailing whitespace.
- `gitignore` updated to match `develop` version (soon to include line related to Codespaces, also included here)
- Docs requirements updated, but docs untouched in this version (outside of `pre-commit` touches)
- AUTHORS updated (take credit for your work!)

### CODE REVIEW

**Review by Bobspadger — COMMENTED** [review]
Removing the makefile - ok, but now hides the build commands for pushing to pypi.

File: mws/future_utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from .collections import unique_list_order_preserved
```
Comment (Bobspadger) [inline mws/future_utils/__init__.py:1]: could there end up with a namespace conflict here, collections is a python standard library?

I've had weird stuff like this happen before, it shouldn't happen, but sometimes does!

**Review by Bobspadger — COMMENTED** [review]
Only comment really is on the .collections namespace.

File: mws/future_utils/__init__.py (hunk 1 above)
Comment (GriceTurrble) [reply mws/future_utils/__init__.py:1]: That is possible, though I think it's less an issue in Python 3 with the absolute imports.

If we're still trying to support Py2 for compatibility in 0.8, then yes that could be a minor issue. Might have to pepper in `from __future__ import absolute_import` to be safe there.

**Comment by GriceTurrble** [comment]
Re: the Makefile removal, I assume that the command `python setup.py sdist bdist_wheel` was used to build and send it over to PyPI?

That's one area where we could have an Actions workflow take over, to trigger on tags been added with new version numbers.

File: mws/future_utils/__init__.py (hunk 1 above)
Comment (Bobspadger) [reply mws/future_utils/__init__.py:1]: > That is possible, though I think it's less an issue in Python 3 with the absolute imports.
> 
> If we're still trying to support Py2 for compatibility in 0.8, then yes that could be a minor issue. Might have to pepper in `from __future__ import absolute_import` to be safe there.

hmmmmm we've pretty much ditched py2 haven't we so don't put more work in than is necessary.

**Comment by Bobspadger** [comment]
> Re: the Makefile removal, I assume that the command `python setup.py sdist bdist_wheel` was used to build and send it over to PyPI?
> 
> That's one area where we could have an Actions workflow take over, to trigger on tags been added with new version numbers.

`python setup.py sdist bdist_wheel` builds it

`python3 -m twine upload dist/*` is what uploads to pypi and requires credentials etc.

**Review by Bobspadger — APPROVED** [review]

**Pull request merged**